### PR TITLE
Fix/notifications delete error

### LINKED ISSUE #20990: [UI] Deleting all notifications shows an error
### Current Behavior
After deleting all notifications, the page shows the error:
`Cannot read properties of null (reading 'find')`

### Expected Behavior
All notifications should be deleted successfully and the page should remain usable without showing an error.

### Steps to Reproduce

1. Open the Notifications section.
2. Make sure there are some notification messages.
3. Click Select All.
4. Click the delete/trash icon.
5. Once all notifications are deleted, check the page.

Note: The issue occurs only when no notification messages remain.
### Screenshots/Logs

https://github.com/user-attachments/assets/9243a21e-5e8f-49b4-812d-6de970cbb452

### Environment

- **Host OS:** Mac Linux Windows
- **Platform:** Docker or Kubernetes
- **Meshery Server Version:** stable-v
- **Meshery Client Version:** stable-v

---
### Contributor [Guides](https://docs.meshery.io/project/contributing) and [Handbook](https://meshery.io/community#handbook)
- 🛠 [Meshery Build & Release Strategy](https://docs.meshery.io/project/contributing/build-and-release)
- 📚 [Instructions for contributing to documentation](https://docs.meshery.io/project/contributing/contributing-docs)
   - Meshery documentation [site](https://docs.meshery.io/) and [source](https://github.com/meshery/meshery/tree/master/docs)
- 🎨 Wireframes and [designs for Meshery UI](https://www.figma.com/file/SMP3zxOjZztdOLtgN4dS2W/Meshery-UI) in Figma [(open invite)](https://www.figma.com/team_invite/redeem/GvB8SudhEOoq3JOvoLaoMs)
- 📺 [Self-paced Contributor Trainings](https://meshery.io/talks-and-trainings#trainings)
- 🙋🏾🙋🏼 Questions: [Discussion Forum](https://meshery.io/community#community-forums) and [Community Slack](https://slack.meshery.io)

### PR DESCRIPTION
**Pull request by Sbragul26**
**Notes for Reviewers**

This PR fixes #20990 by handling the empty notifications state after deleting all notifications.

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [ ] Yes, I signed my commits.

## Summary by CodeRabbit

- **Bug Fixes**
  - Improved notification severity counts when severity data is unavailable.
  - Displays a count of zero instead of causing an error when no severity counts are provided.
  - Notification Center now remains stable when summary information or severity details are missing.
- **Tests**
  - Added coverage to verify that all severity counts display as zero when no severity data is returned.

### CODE REVIEW

**Comment by Sbragul26** [comment]
**Preview:**


https://github.com/user-attachments/assets/ead2de99-a2ac-415e-a631-ca0c9158cdeb

**Comment by dhruveshmishra** [comment]
@Sbragul26 I think for this issue pr has been raised #20859

**Comment by Sbragul26** [comment]
@dhruveshmishra I think they’re related but not the same issue. #20859 is about the UI not updating correctly after all notifications are deleted, while #20990 is about an error occurring when the notification severity/count data is empty. This PR specifically fixes the error in that empty state.

File: ui/components/layout/NotificationCenter/index.tsx [hunk 1]
```diff
@@ -97,7 +97,8 @@ export const NotificationCenterProvider = ({ children }) => {
 };
 
 const getSeverityCount = (countBySeverityLevel, severity) => {
-  return countBySeverityLevel.find((item) => item.severity === severity)?.count || 0;
+  const severityLevels = countBySeverityLevel || [];
+  return severityLevels.find((item) => item.severity === severity)?.count || 0;
```
[the remark this answers is not in the document]
Comment (Sbragul26) [reply ui/components/layout/NotificationCenter/index.tsx:101]: Addressed by adding a test for `countBySeverityLevel: null`.
